### DISCUSSION TITLE
ERR_TLS_CERT_ALTNAME_INVALID

### DISCUSSION THREAD

**Discussion by coderpradp** [chimurai/http-proxy-middleware#557] [Q&A]
### Expected behavior
I'm supposed to get the request via proxy
### Actual behavior
I'm getting ALTNAME error
### Setup

* http-proxy-middleware: 0.17.4
* server: express + 4.15.2
* other relevant modules

#### proxy middleware configuration
```javascript
function relayRequestHeaders(proxyReq, req) {
    Object.keys(req.headers).forEach(function (key) {
        proxyReq.setHeader(key, req.headers[key]);
    });
};

function relayResponseHeaders(proxyRes, req, res) {
    Object.keys(proxyRes.headers).forEach(function (key) {
        res.append(key, proxyRes.headers[key]);
    });
};

const options = {
	target: 'example.com',
	logLevel: 'debug',
	onProxyReq: relayRequestHeaders,
	onProxyRes: relayResponseHeaders
}
```
#### server mounting
```javascript
var app = express();

app.use('/apps', proxy(options));
app.listen(3000);
```

**Comment by coderpradp** [comment]
When ever I add `changeOrigin: true` I get 404 errors. Any help?

**Comment by robertwbradford** [comment]
@coderpradp Did you ever figure this out?

**Comment by ernitishkumar** [comment]
Getting same error when proxying request to a ssl based server

**Comment by alexichepura** [comment]
Had same issue. 
Added `changeOrigin` to fix this and `pathRewrite` to fix 404
```js
  const apiProxy = proxy('/api', {
    target: API_ENDPOINT,
    changeOrigin: true,
    pathRewrite: { '^/api': '' }, // remove leading /api to match real API urls
  })
```

**Reply by KKrisu** [reply]
perfect, thank you @alexichepura

**Comment by faizan1990** [comment]
What location is this file found at?

**Comment by dardino** [comment]
I have already set `changeOrigin` to true.
I think that the error came from a *self signed* certificate.
then:
how to accept SELF SIGNED certificate in dev environment?

**Comment by dardino** [comment]
> I have already set `changeOrigin` to true.
> I think that the error came from a _self signed_ certificate.
> then:
> how to accept SELF SIGNED certificate in dev environment?

it seems to be enough to set `secure: false`

**Reply by bhushan-patil-official** [reply]
This solution works while I am accessing a secure ip from the localhost using `http-proxy-middleware` although, is there anyway we can provide the self signed ssl certificates to from the frontend to bypass it?

**Reply by chimurai** [reply]
You can try the `ssl` option.

Here is an example:
https://github.com/http-party/node-http-proxy/blob/master/examples/http/proxy-https-to-http.js#L53-L56

**Comment by webunderconstruction** [comment]
A test https server I was trying to proxy to was using a self signed certificate, I found that by striping all the request headers seemed to allow it to successfully connect and return a response.

I changed the `relayRequestHeaders` function to this;

```javascript
function relayRequestHeaders(proxyReq, req) {
  Object.keys(req.headers).forEach(function (key) {
    // strip all the request headers, as the receiving server is freaking out on one of them
    proxyReq.setHeader(key, '');
  });
}
```
